explorer-api/src/mix_nodes/mod.rs

```diff
@@ -21,12 +41,42 @@ impl ThreadsafeMixNodesResult {
     pub(crate) fn new() -> Self {
         ThreadsafeMixNodesResult {
             inner: Arc::new(RwLock::new(MixNodesResult {
-                value: vec![],
+                value: HashMap::new(),
                 valid_until: SystemTime::now() - Duration::from_secs(60), // in the past
+                location_cache: LocationCache::new(),
             })),
         }
     }
 
+    pub(crate) fn attach(location_cache: LocationCache) -> Self {
```

Comment:
I would rename this to `new_with_location_cache`, as attaching looks like doing work on something already existing.

Reply:
Yep. I agree, `attach` seems like something that should have been called on `self`

Reply:
Perhaps you might change it into `pub(crate) fn attach(mut self, location_cache: LocationCache) -> Self` method that only modified the `location_cache` so that you could call `ThreadsafeMixNodesResult::new().attach(cache)`

Reply:
Giving it a shot with `suggestion`
```suggestion
    pub(crate) fn new_with_location_cache(location_cache: LocationCache) -> Self {
```